scripts/utils.py

```diff
@@ -177,3 +177,12 @@ def get_requirements_from_poetry(
         requirements.append(requirement_line)
 
     return "\n".join(requirements)
+
+
+def normalize(dependency_name: str) -> str:
+    """
+    For consistent comparisons between pyproject.toml and lockfile,
+    normalize names consistent with PEP503:
+    https://peps.python.org/pep-0503/#normalized-names
```

Comment:
Nit: I found it helpful to understand that this is a blessed, self-sufficient implementation of this normalization:

```suggestion
    normalize names consistent with PEP503.  This implementation is drawn directly from
    <https://peps.python.org/pep-0503/#normalized-names>.
```